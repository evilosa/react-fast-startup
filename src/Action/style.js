// @flow
export const ActionColors = {

};

export const createActionStyle = (colors = ActionColors) => ({
  colors,
});

export const ActionStyle = createActionStyle(ActionColors);

export default ActionStyle;