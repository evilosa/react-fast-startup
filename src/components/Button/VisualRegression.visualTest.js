import { Chrome } from 'navalia';
import { toMatchImageSnapshot } from 'jest-image-snapshot';

expect.extend({ toMatchImageSnapshot });

describe('Visual Regressions', () => {
  let chrome = null;
  let server = null;

  beforeEach(() => {
    server = require('../../../scripts/server');
    chrome = new Chrome({
      flags: {
        headless: true,
        disableGpu: true,
        hideScrollbars: true,
        noSandbox: true
      }
    });
  });

  afterEach(() => {
    chrome.done();
    server.close();
  });

  it('should NEVER happen', () => {
    return chrome.goto('http://localhost:3000/')
      .then(() => chrome.screenshot())
      .then((image) => {
        console.log(`Image length: ${image.length}`);
        expect(image).toMatchImageSnapshot();
      });
  });
});