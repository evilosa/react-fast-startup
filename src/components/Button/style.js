import Color from 'color';

export const ButtonColors = {
  baseColor: '#fff',
  baseBackground: '#222222',
  primaryBackground: '#0074d9',
  warningBackground: '#f00',
};

export const createButtonStyle = (colors = ButtonColors) => ({
  base: {
    color: colors.baseColor,
    ':hover': {
      background: Color(colors.baseBackground).lighten(0.2).hex(),
    },
    background: colors.baseBackground,
  },

  primary: {
    background: colors.baseBackground,
  },

  small: {
    fontSize: '12px',
    padding: '5px',
    fontWeight: '300',
    borderRadius: '4px',
  },

  normal: {
    fontSize: '16px',
    fontWeight: '300',
    padding: '10px',
    borderRadius: '4px',
  },

  big: {
    fontSize: '24px',
    fontWeight: '300',
    padding: '15px',
    minWidth: '150px',
    borderRadius: '4px',
  },

  warning: {
    background: colors.warningBackground,
  }
});

export const ButtonStyle = createButtonStyle(ButtonColors);

export default ButtonStyle;
