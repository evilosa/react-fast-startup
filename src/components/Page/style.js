export const PageColors = {
  color: '#fff',
  background: '#222222',
};

export const createPageStyle = (/*colors = PageColors*/) => ({
  root: {
    display: 'flex',
    flexDirection: 'column',
    flex: '1 0',
    width: '100%',
    height: '100%',
  },

  header: {
    display: 'flex',
    flex: '1 0',
    height: 'auto',
  },

  content: {
    display: 'flex',
    flexDirection: 'column',
    flex: '1 0',
  },

  footer: {
    display: 'flex',
    flex: '1 0',
    height: 'auto',
  }
});

export const ButtonStyle = createPageStyle();

export default ButtonStyle;
