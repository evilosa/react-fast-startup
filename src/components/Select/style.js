export const SelectColors = {
  color: '#fff',
  rootBorder: '#3b88ff',
  titleBorder: '#ff6f13',
  valueBorder: '#5bff42',
  optionsBorder: '#fcff19',
  searchBorder: '#ff0019',
  background: '#222222',
};

const defaultBorder = {
  borderWidth: '1px',
  borderStyle: 'solid',
}

export const createSelectStyle = (colors = SelectColors) => ({
  root: {
    display: 'flex',
    flex: '1 0',
    ...defaultBorder,
    borderColor: colors.rootBorder,
  },

  title: {
    display: 'flex',
    padding: '5px',
    alignItems: 'center',
    minWidth: '150px',
    ...defaultBorder,
    borderColor: colors.titleBorder,
  },

  value: {
    flex: '1 0',
    padding: '5px',
    ...defaultBorder,
    borderColor: colors.valueBorder,
  },

  optionsList: {
    display: 'block',
    position: 'absolute',
    backgroundColor: '#f9f9f9',
    maxHeight: '60px',
    minWidth: '160px',
    boxShadow: '0px 8px 16px 0px rgba(0,0,0,0.2)',
    marginTop: '30px',
    padding: '12px 16px',
    zIndex: '1',
    overflowY: 'auto',
    overflowX: 'hidden',
  },
});

export const SelectStyle = createSelectStyle();

export default SelectStyle;
