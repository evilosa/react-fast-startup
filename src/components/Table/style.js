export const TableColors = {
  baseBackground: '#fff',
  evenBackground: '#EEF3F5',
  border: '#D7DBDD',
  headerBackground: '#FCFDFD',
  headerText: '#16A2D7'
}

export const createTableStyle = (colors = TableColors) => ({
  root: {
    display: 'flex',
    flexDirection: 'column',
    flex: '1 0',
  },

  header: {
    display: 'flex',
    flex: '1 0',
    backgroundColor: colors.headerBackground,
    minHeight: '30px',
    maxHeight: '30px',
    borderBottomWidth: '1px',
    borderBottomStyle: 'solid',
    borderBottomColor: colors.border
  },

  body: {
    display: 'flex',
    flexDirection: 'column',
    flex: '1 0',
  },

  footer: {
    display: 'flex',
    flex: '1 0',
    minHeight: '30px',
    maxHeight: '30px',
  },

  tableRow: {
    display: 'flex',
    flex: '1 0',
    background: colors.baseBackground,
    minHeight: '30px',
    maxHeight: '30px',
  },

  even: {
    background: colors.evenBackground,
  },

  tableHeaderColumn: {
    display: 'flex',
    flex: '1 0',
    padding: '5px',
    justifyContent: 'flex-start',
    alignItems: 'center',
    fontFamily: "'Raleway', sans-serif",
    fontSize: '12px',
    fontWeight: '400',
    lineHeight: '18px',
    fontStretch: '100%',
    color: colors.headerText,
  },

  tableRowActionColumn: {
    flex: '1 0',
    padding: '5px',
    fontFamily: "'Raleway', sans-serif",
    fontSize: '12px',
    fontWeight: '400',
    lineHeight: '18px',
    fontStretch: '100%',
  },

  tableRowColumn: {
    flex: '1 0',
    padding: '5px',
    fontFamily: "'Raleway', sans-serif",
    fontSize: '12px',
    fontWeight: '400',
    lineHeight: '18px',
    fontStretch: '100%',
  }
})

export const TableStyle = createTableStyle(TableColors)

export default TableStyle
