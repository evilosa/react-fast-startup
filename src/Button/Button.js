// @flow
import * as React from 'react';
import PropTypes from 'prop-types';
import Radium from 'radium';
import style from './style';

class Button extends React.Component {

  static propTypes = {
    /** Default style for button */
    style: PropTypes.object,
  };

  static defaultProps = {
    style,
  };

  _onClick = () => {
    const { style } = this.props;
    console.log('Button style');
    console.log(style);
  };

  render() {
    const { style, inlineStyle, children } = this.props;

    return (
      <button style={[style.base, style.primary, inlineStyle]} onClick={this._onClick} type="Button">{children}</button>
    )
  }
}

//$FlowFixMe
export default Radium(Button);