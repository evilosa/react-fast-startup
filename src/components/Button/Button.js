// @flow
import * as React from 'react';
import PropTypes from 'prop-types';
import Radium from 'radium';
import style from './style';

class Button extends React.Component {

  static propTypes = {
    /** Default style for button */
    style: PropTypes.object,
    type: PropTypes.string,
    onClick: PropTypes.func
  };

  static defaultProps = {
    style,
    type: 'normal',
  };

  _onClick = () => {};

  render() {
    const { style, onClick, inlineStyle, children, type } = this.props;

    return (
      <button style={[style.base, style.primary, style[type], inlineStyle]} onClick={onClick} type="Button">{children}</button>
    )
  }
}

//$FlowFixMe
export default Radium(Button);