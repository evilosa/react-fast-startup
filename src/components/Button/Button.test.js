import React from 'react'
import { shallow } from 'enzyme'
import Button from './Button'
import { createButtonStyle } from './style';

describe('Button', () => {
  const button = shallow(<Button/>)

  it('renders correct', () => {
    expect(button).toMatchSnapshot()
  })

  it('should call provided function', () => {
    const mockCallBack = jest.fn()

    const button = shallow(<Button onClick={mockCallBack}>Ok!</Button>)
    button.find('button').simulate('click')
    expect(mockCallBack.mock.calls.length).toEqual(1)
  })

  it('should have correct style', () => {
    expect(createButtonStyle()).toMatchSnapshot();
  })
})