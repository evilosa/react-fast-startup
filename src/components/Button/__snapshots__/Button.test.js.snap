// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders correct 1`] = `
<button
  data-radium={true}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  style={
    Object {
      "background": "#222222",
      "borderRadius": "4px",
      "color": "#fff",
      "fontSize": "16px",
      "fontWeight": "300",
      "padding": "10px",
    }
  }
  type="Button"
/>
`;

exports[`Button should have correct style 1`] = `
Object {
  "base": Object {
    ":hover": Object {
      "background": "#292929",
    },
    "background": "#222222",
    "color": "#fff",
  },
  "big": Object {
    "borderRadius": "4px",
    "fontSize": "24px",
    "fontWeight": "300",
    "minWidth": "150px",
    "padding": "15px",
  },
  "normal": Object {
    "borderRadius": "4px",
    "fontSize": "16px",
    "fontWeight": "300",
    "padding": "10px",
  },
  "primary": Object {
    "background": "#222222",
  },
  "small": Object {
    "borderRadius": "4px",
    "fontSize": "12px",
    "fontWeight": "300",
    "padding": "5px",
  },
  "warning": Object {
    "background": "#f00",
  },
}
`;
